src/ngraph/pass/manager.hpp

```diff
@@ -62,20 +61,30 @@ class ngraph::pass::Manager
             m_pass_names.push_back(typeid(T).name());
 #endif
         }
+        if (m_per_pass_validation && !std::is_same<T, Validate>::value)
```

Comment:
It's not a big deal, but I wonder if this is overkill.

Reply:
I was wondering the same thing too. I doubt that users will use the "auto insert validation" mode along with adding Validate passes into the graph. I will remove the `is_same`. Double validation never killed anyone.